### ISSUE TITLE
5.x: Asset compilation with sprockets is broken on JRuby on Windows

### ISSUE THREAD

**Issue by bartkamphorst** [gollum/gollum#1321]
```
$ gollum --versions
Gollum 5.0.1b
Running on: java with Ruby version 2.5.0
Using:
rjgit 5.1.1.0
gollum-rjgit_adapter 0.3.4
gollum-lib 5.0.a.4
With the following renderers:
kramdown 1.9.0
```
```
ExecJS::RuntimeError: SyntaxError: Unexpected token: name (b)
                     extract_result at C:/jruby-9.2.0.0/lib/ruby/gems/shared/gems/execjs-2.7.0/lib/execjs/external_runtime.rb:85
                               exec at C:/jruby-9.2.0.0/lib/ruby/gems/shared/gems/execjs-2.7.0/lib/execjs/external_runtime.rb:39
                               eval at C:/jruby-9.2.0.0/lib/ruby/gems/shared/gems/execjs-2.7.0/lib/execjs/external_runtime.rb:21
                               call at C:/jruby-9.2.0.0/lib/ruby/gems/shared/gems/execjs-2.7.0/lib/execjs/external_runtime.rb:46
                       run_uglifyjs at C:/jruby-9.2.0.0/lib/ruby/gems/shared/gems/uglifier-3.2.0/lib/uglifier.rb:195
                            compile at C:/jruby-9.2.0.0/lib/ruby/gems/shared/gems/uglifier-3.2.0/lib/uglifier.rb:157
            block in js_compressor= at C:/jruby-9.2.0.0/lib/ruby/gems/shared/gems/sprockets-3.7.2/lib/sprockets/compressing.rb:65
                               call at C:/jruby-9.2.0.0/lib/ruby/gems/shared/gems/sprockets-3.7.2/lib/sprockets/legacy_proc_processor.rb:31
                     call_processor at C:/jruby-9.2.0.0/lib/ruby/gems/shared/gems/sprockets-3.7.2/lib/sprockets/processor_utils.rb:75
           block in call_processors at C:/jruby-9.2.0.0/lib/ruby/gems/shared/gems/sprockets-3.7.2/lib/sprockets/processor_utils.rb:57
                       reverse_each at org/jruby/RubyArray.java:1885
                    call_processors at C:/jruby-9.2.0.0/lib/ruby/gems/shared/gems/sprockets-3.7.2/lib/sprockets/processor_utils.rb:56
                 load_from_unloaded at C:/jruby-9.2.0.0/lib/ruby/gems/shared/gems/sprockets-3.7.2/lib/sprockets/loader.rb:134
                      block in load at C:/jruby-9.2.0.0/lib/ruby/gems/shared/gems/sprockets-3.7.2/lib/sprockets/loader.rb:60
  fetch_asset_from_dependency_cache at C:/jruby-9.2.0.0/lib/ruby/gems/shared/gems/sprockets-3.7.2/lib/sprockets/loader.rb:317
                               load at C:/jruby-9.2.0.0/lib/ruby/gems/shared/gems/sprockets-3.7.2/lib/sprockets/loader.rb:44
                block in initialize at C:/jruby-9.2.0.0/lib/ruby/gems/shared/gems/sprockets-3.7.2/lib/sprockets/cached_environment.rb:20
                            default at org/jruby/RubyHash.java:737
                                 [] at org/jruby/RubyHash.java:1150
                               load at C:/jruby-9.2.0.0/lib/ruby/gems/shared/gems/sprockets-3.7.2/lib/sprockets/cached_environment.rb:47
                         find_asset at C:/jruby-9.2.0.0/lib/ruby/gems/shared/gems/sprockets-3.7.2/lib/sprockets/base.rb:66
             find_all_linked_assets at C:/jruby-9.2.0.0/lib/ruby/gems/shared/gems/sprockets-3.7.2/lib/sprockets/base.rb:73
                      block in find at C:/jruby-9.2.0.0/lib/ruby/gems/shared/gems/sprockets-3.7.2/lib/sprockets/manifest.rb:134
                               each at org/jruby/RubyArray.java:1801
                               find at C:/jruby-9.2.0.0/lib/ruby/gems/shared/gems/sprockets-3.7.2/lib/sprockets/manifest.rb:133
                            compile at C:/jruby-9.2.0.0/lib/ruby/gems/shared/gems/sprockets-3.7.2/lib/sprockets/manifest.rb:186
                             <main> at C:/Users/gollum/gollum/bin/gollum:276
                               load at org/jruby/RubyKernel.java:994
                             <main> at C:\jruby-9.2.0.0\lib\ruby\gems\shared\bin\gollum:23
```

**Comment by dometto** [comment]
Can't remember, but was precompilation failing only on Windows, or JRuby in general? After [this fix](https://github.com/gollum/gollum/pull/1338/files#diff-2dc6aed30f72635fc751fe61e07c11d4R271) in #1338 `--precompile` is working on JRuby on OS X at least. Maybe the problem was just that precompile was broken all the time?

**Comment by bartkamphorst** [comment]
> was precompilation failing only on Windows, or JRuby in general?

JRuby on Windows. JRuby on OS X was working fine before.

**Comment by bartkamphorst** [comment]
@dometto The `Unexpected token: name (b)` refers to the name variable in this helper method:
```ruby
  def sprockets_javascript_tag
    lambda do |name|
      send(:javascript_tag, name)
    end
  end
```

Could you explain to me what this is suppose to be doing?

**Comment by dometto** [comment]
It's a way to utilize sprockets tag helpers in mustache (in layout.mustache). I think mustache expects `{{#bla}}argument{{/bla}}` to return a block, to which `argument` is passed.

**Comment by bartkamphorst** [comment]
Right, but it boggles me why `execjs` would complain about this.

**Comment by dometto** [comment]
Me too! Could opt for a different notation for the block and see if that makes a difference, but this is weird.

**Comment by dometto** [comment]
Tried updating to the `sassc` gem to see if it was better, but there are some problems in compatibility with sprockets, that will hopefully be fixed when sprockets 4.0 is released soon. Issue to watch:

* https://github.com/rails/sprockets/issues/565

**Comment by ahorek** [comment]
just a guess, but do you have a js runtime installed? https://github.com/cowboyd/therubyrhino or https://nodejs.org

**Comment by bartkamphorst** [comment]
@ahorek Thanks for pitching in! Adding therubyrhino as a runtime dependency for JRuby might just do the trick. If you know anyone who uses Windows who could confirm this for us, that would be great. 👍

**Comment by dometto** [comment]
Only if we want to offer the option of running gollum without static assets when not using `bundle exec` -- so basically, if we want to offer the option of running gollum without static assets in production.

**Comment by dometto** [comment]
> Only if we want to offer the option of running gollum without static assets when not using bundle exec -- so basically, if we want to offer the option of running gollum without static assets in production.

Ok to leave the dependency in the Gemfile for now?

**Comment by bartkamphorst** [comment]
> Ok to leave the dependency in the Gemfile for now?

Sure.

**Comment by dometto** [comment]
Fixed #1425
